### PR TITLE
Fix runtime stats when cache hit

### PR DESCRIPTION
**Pull request by beinan**
### What changes are proposed in this pull request?
Fix runtime stats when cache hit

### Why are the changes needed?

We already count the runtime stats in local cache manager, but we didn't pass in a proper cache context

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
  1. change in user-facing APIs
  2. addition or removal of property keys
  3. webui

### CODE REVIEW

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java [hunk 1]
```diff
@@ -927,10 +929,20 @@ private int getPage(PageInfo pageInfo, int pageOffset, int bytesToRead,
         // data read from page store is inconsistent from the metastore
         LOG.error("Failed to read page {}: supposed to read {} bytes, {} bytes actually read",
             pageInfo.getPageId(), bytesToRead, ret);
+        target.offset(originOffset); //reset the offset
+        //best efforts to delete the corrupted file without acquire the write lock
+        deletePage(pageInfo, false);
```
Comment (JiamingMai) [inline dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java:934]: Do we need to delete the metadata as well?

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java [hunk 2]
```diff
@@ -927,10 +929,20 @@ private int getPage(PageInfo pageInfo, int pageOffset, int bytesToRead,
...
         return -1;
       }
+    } catch (PageCorruptedException e) {
+      LOG.error("Data corrupted page {} from pageStore", pageInfo.getPageId(), e);
+      target.offset(originOffset); //reset the offset
+      //best efforts to delete the corrupted file without acquire the write lock
+      deletePage(pageInfo, false);
```
Comment (JiamingMai) [inline dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java:941]: Do we need to delete the metadata as well?

**Review by JiamingMai — APPROVED** [review]
LGTM. Just left a comment. If we read a corrupted page, do we need to delete the metadata as well?

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java (hunk 1 above)
Comment (beinan) [reply dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java:934]: Deleting the metadata here will acquire more locks, which might cause dead lock.  So I just delete the page file, and page store will re-cache this page in the next visit.

Comment (beinan) [reply dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java:934]: I think it's a good call, probably we can have a further improvement in the near future. But just for now, ,et's merge this pr and stop the bleeding of the data corruption issue

**Comment by gggrace14** [comment]
Hi @beinan , this fix works from our testing. Would it be possible to make it into version 308?

**Comment by beinan** [comment]
alluxio-bot, merge this please.

**Pull request merged**